vault-core/src/main/java/org/apache/jackrabbit/vault/packaging/impl/JcrPackageManagerImpl.java

```diff
@@ -88,8 +88,8 @@
      * @param session repository session
      * @param roots the root paths to store the packages.
      */
-    public JcrPackageManagerImpl(Session session, String[] roots) {
-        this(new JcrPackageRegistry(session, roots));
+    public JcrPackageManagerImpl(@NotNull Session session, @Nullable String[] roots, @Nullable String[] additionalAuthorizableIdsAllowedToExecuteHooks, @Nullable String[] additionalAuthorizableIdsAllowedToInstallPackagesRequiringRoot) {
+        this(new JcrPackageRegistry(session, additionalAuthorizableIdsAllowedToExecuteHooks, additionalAuthorizableIdsAllowedToInstallPackagesRequiringRoot, roots));
```

Comment:
I don't like this in particular:

- what about creating a `JcrPackageRegistryConfig` that holds the `roots`, `authIdsExec` and `authIdsRootInstall` value?
- or at least creating a `JcrPackageRegistrySecurity` config that holds the `authIdsExec` and `authIdsRootInstall` values?

in any case: also add default constructors that implement the old api.

Reply:
For me this is not API, as this is in a not exported impl package, so keeping backwards-compatibility is not necessary as this can only be used by the bundle itself. But wrapping the config makes sense.

Reply:
it's not about backward compatibility, but to write `...null, null)` alot is kind of disturbing.